brewtils/decorators.py

```diff
@@ -358,154 +302,239 @@ def cmd1(self, **kwargs):
+    based on the function signature.
+
+    Regardless, we went with this originally. If we want to change it we need to go
+    though a deprecation cycle and *loudly* publicize it since things wouldn't break
+    loudly for plugin developers, their plugins would just be subtly (but importantly)
+    different.
+
+    Args:
+        method: The method with the Command to initialize
+
+    Returns:
+        The initialized Command
 
-    The main thing this does is ensure the choices specification is correct for all
-    Parameters in the tree.
     """
-    parameters_to_return = []
+    cmd = getattr(method, "_command", Command())
 
-    for param in parameter_list:
+    cmd.name = _method_name(method)
+    cmd.description = cmd.description or _method_docstring(method)
 
-        # This is already a Parameter. Only really need to interpret the choices
-        # definition and recurse down into nested Parameters
-        if isinstance(param, Parameter):
-            new_param = Parameter(
-                key=param.key,
-                type=param.type,
-                multi=param.multi,
-                display_name=param.display_name,
-                optional=param.optional,
-                default=param.default,
-                description=param.description,
-                choices=_format_choices(param.choices),
-                nullable=param.nullable,
-                maximum=param.maximum,
-                minimum=param.minimum,
-                regex=param.regex,
-                type_info=param.type_info,
-            )
+    resolved_mod = _resolve_display_modifiers(
+        method, cmd.name, schema=cmd.schema, form=cmd.form, template=cmd.template
+    )
+    cmd.schema = resolved_mod["schema"]
+    cmd.form = resolved_mod["form"]
+    cmd.template = resolved_mod["template"]
 
-            if param.parameters:
-                new_param.type = "Dictionary"
-                new_param.parameters = _generate_nested_params(param.parameters)
+    # Now we need to deal with parameters. Start with ones from @parameter decorators.
+    cmd.parameters += getattr(method, "parameters", [])
 
-            parameters_to_return.append(new_param)
+    # Now we need to reconcile the parameters with the method signature
+    for index, arg in enumerate(signature(method).parameters.values()):
 
-        # This is a model class object. Needed for backwards compatibility
-        # See https://github.com/beer-garden/beer-garden/issues/354
-        elif hasattr(param, "parameters"):
-            _deprecate(
-                "Constructing a nested Parameters list using model class objects "
-                "is deprecated. Please pass the model's parameter list directly."
+        # Don't want to include special parameters
+        if index == 0 and arg.name in ("self", "cls"):
+            continue
+
+        # Grab default and optionality according to the signature. We'll need it later.
+        sig_default, sig_optional = _sig_info(arg)
```

Comment:
I love this whole for loop. I was able to write a ticket when I started this PR because I didn't realize it was in here. I think we need to add `sig_type` to your function here. We can default to String, but if they use a primitive type then lets represent that in the UI.

Reply:
This loop also makes it so if they are using primitive types, then they only need the `@command` over their function and it just works.

Reply:
Ohhh interesting... you're saying that for something like

```python
@command
def foo(x=3):
    pass
```

we should be able to say that the type for parameter `x` is Integer?

We'll have to think through all the implications of that. But I like it.